aqueduct/gateway/tests/test_mcp.py

```diff
@@ -0,0 +1,712 @@
+import logging
+from datetime import timedelta
+
+from django.conf import settings
+from pydantic.networks import AnyUrl
+
+from gateway.tests.utils.mcp import MCPLiveServerTestCase
+
+if settings.TESTING:
+    logger = logging.getLogger("aqueduct")
+    logging.disable(logging.NOTSET)
+    logger.setLevel(logging.DEBUG)
+
+
+class MCPLiveClientTest(MCPLiveServerTestCase):
+    async def test_list_tools(self):
+        """Test MCP tools listing."""
+        async with self.client_session() as session:
+            await session.initialize()
+            tools = await session.list_tools()
+
+            self.assertIsNotNone(tools)
+            self.assertIsInstance(tools.tools, list)
+            self.assertGreater(len(tools.tools), 0)
+            # Verify expected tools are present
+            tool_names = [tool.name for tool in tools.tools]
+            self.assertIn("echo", tool_names)
+
+        await self.assertRequestLogged()
+
+    async def test_call_tool(self):
+        """Test calling a tool."""
+        async with self.client_session() as session:
+            await session.initialize()
+            tool_name = "echo"
+            result = await session.call_tool(tool_name, {"message": "test"})
+
+            self.assertIsNotNone(result)
+            self.assertIsInstance(result.content, list)
+            self.assertGreater(len(result.content), 0)
+            # Verify the echo tool returns the input message
+            self.assertIn("test", result.content[0].text)
+
+        await self.assertRequestLogged()
+
+    async def test_call_tool_long_running(self):
+        """Test calling a long-running tool with progress updates."""
+        async with self.client_session() as session:
+            await session.initialize()
+            tool_name = "longRunningOperation"
+            result = await session.call_tool(tool_name, {"duration": 3, "steps": 5})
+
+            self.assertIsNotNone(result)
+            self.assertIsInstance(result.content, list)
+            self.assertGreater(len(result.content), 0)
```

Comment:
Do we test anything related to progress updates here? Is there any way to do it? Otherwise this test doesn't differ much from the `test_call_tool`, and as such seems redundant...?